### PR TITLE
Fix minimal-apis.md's TOC

### PR DESCRIPTION
**Pull request by halter73**
This PR moves the [Bind the request body as a Stream or PipeReader](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0#bind-the-request-body-as-a-stream-or-pipereader), [Parameter binding for argument lists with [AsParameters]](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0#parameter-binding-for-argument-lists-with-asparameters), [File uploads using IFormFile and IFormFileCollection](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0#file-uploads-using-iformfile-and-iformfilecollection) from the top of the TOC to under "[Request Handling](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0#request-handling-1) > [ParameterBinding](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0#parameter-binding-1)" where they belong.

**EDIT** Also.
- Moved the "Typed Results" section under "Responses"
- Nested example IResult return values under the "IResult return values" section
- Briefly mention `TypedResults` in the "Typed Results" section

#### 6.0 TOC

image

#### 7.0 TOC (Current)

image

#### 7.0 TOC (With PR)

image

[Internal Review link](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-7.0&branch=pr-en-us-27329)

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
Regarding the changes in the Responses section, we now have an [article devoted to responses](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/responses?view=aspnetcore-7.0), and instead of tweaking the Responses section here, it might make sense to remove the Responses section from this article and replace it with a link to the Responses article. We'd just have to verify that nothing here is missing from that article.

File: aspnetcore/fundamentals/minimal-apis.md [hunk 1]
```diff
@@ -1376,6 +1297,44 @@ The following types are bound without explicit attributes:
   app.MapGet("/", (ClaimsPrincipal user) => user.Identity.Name);
   ```
 
+<a name="rbs"></a>
+
+#### Bind the request body as a `Stream` or `PipeReader`
+
+The request body can bind as a [`Stream`](/dotnet/api/system.io.stream) or [`PipeReader`](/dotnet/api/system.io.pipelines.pipereader) to efficiently support scenarios where the user has to process data and:
```
Comment (captainsafia) [inline aspnetcore/fundamentals/minimal-apis.md:1304]: This feels like something we want to pull out and expand on under a separate article.

File: aspnetcore/fundamentals/minimal-apis.md [hunk 2]
```diff
@@ -1620,6 +1620,16 @@ We recommend adding an extension method to <xref:Microsoft.AspNetCore.Http.IResu
 
 [!code-csharp[](minimal-apis/7.0-samples/WebMinAPIs/Program.cs?name=snippet_xtn)]
 
+### Typed results
+
+The <xref:Microsoft.AspNetCore.Http.IResult> interface can represent values returned from minimal APIs that don't utilize the implicit support for JSON serializing the returned object to the HTTP response. The static [Results](/dotnet/api/microsoft.aspnetcore.http.results) class is used to create varying `IResult` objects that represent different types of responses. For example, setting the response status code or redirecting to another URL.
+
+The types implementing `IResult` are public, allowing for type assertions when testing. For example:
```
Comment (captainsafia) [inline aspnetcore/fundamentals/minimal-apis.md:1627]: I think it makes sense to link to the new response's page that @brunolins16 created here.

**Review by captainsafia — APPROVED** [review]
Changing of the heading seems fine.

Some of the content prose that were added for .NET 7 should be stripped down a little bit. Most of the cheatsheet is headings + code samples so it's a little jarring to have content that is prose + code samples. It probably makes sense to pull these out into separate articles.

**Comment by Rick-Anderson** [comment]
> Regarding the changes in the Responses section, we now have an [article devoted to responses](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/responses?view=aspnetcore-7.0), and instead of tweaking the Responses section here, it might make sense to remove the Responses section from this article and replace it with a link to the Responses article. We'd just have to verify that nothing here is missing from that article.

@captainsafia Isn't it better to remove duplication and have one more complete section on responses?

**Pull request merged**